edx/edx-ora2
# feat: add lock status to submission lock responses

**TL;DR -** Add `lock_status` to submission lock responses.

JIRA: Supporting work in [AU-366](https://openedx.atlassian.net/browse/AU-366)
Linked PR: TBD

**What changed?**

- Added `lock_status` to `SubmissionLockSerializer ` and all submission lock responses.
- Updated response for absent/expired submission lock. Returns `lock_status: 'unlocked'` instead of empty dict.
- Updated docstrings.

**Developer Checklist**
n/a

**Testing Instructions**

Exercise the following endpoints and test-cases (NOTE - all requests require a valid `X-CSRFToken` header).

Check submission lock:
- `POST {{protocol}}://{{lms_url}}/courses/{{course_id}}/xblock/{{block_id}}/handler/check_submission_lock`

Claim a submission lock:
- `POST {{protocol}}://{{lms_url}}/courses/{{course_id}}/xblock/{{block_id}}/handler/claim_submission_lock`

Delete submission lock:
- `POST {{protocol}}://{{lms_url}}/courses/{{course_id}}/xblock/{{block_id}}/handler/delete_submission_lock`

Expected return values:
```
Success: HTTP 200 {
    "submission_uuid": <submisison_uuid or empty>,
    "owner_id": <anonymous_user_id or empty>,
    "created_at": <timestamp or empty>,
    "lock_status": <"unlocked", "in-progress", or "locked">
}

Contested lock: HTTP 403 {
    "error": <error message>
}
```
**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:


FYI: @edx/masters-devs-gta
